src/mgr/ServiceMap.cc

```diff
@@ -75,20 +76,105 @@ std::string ServiceMap::Service::get_summary() const
   if (daemons.empty()) {
     return "no daemons active";
   }
+
   std::ostringstream ss;
-  ss << daemons.size() << (daemons.size() > 1 ? " daemons" : " daemon")
-     << " active";
+  /*
+   * The format is "type/prefix:daemon_name".
+   * "type": will be used to replace the default "daemon(s)"
+   * showed in `ceph -s`. If absent, the "daemon" will used.
+   * "prefix": if present the active members will be classified
+   * by the "prefix" instead of "daemon_name".
+   *
+   * For exmaple for iscsi gateways, it will be:
+   * "portal/hostX:rbd/imageX"
+   * The `ceph -s` will be:
+   *     igw: 3 portals active (hostX, hostY, hostZ)
+   */
+  std::set<std::string> prefs;
+  bool classified = false;
+  auto p = daemons.begin();
+  std::regex pattern("(\\w*)/(\\w*):(\\S+)"); // full pattern
+  std::smatch match;
+  std::string type;
+  if (std::regex_search(p->first, match, pattern)) {
+    type = std::string(match[1]);
+    std::string prefix = std::string(match[2]);
```

Comment:
what about

s/prefix/first_prefix/
s/name/first_name/

?

Reply:
Looks better and will fix it.